src/Illuminate/Routing/RouteDependencyResolverTrait.php

```diff
@@ -74,19 +87,68 @@ public function resolveMethodDependencies(array $parameters, ReflectionFunctionA
      * @param  array  $originalParameters
      * @return mixed
      */
-    protected function transformDependency(ReflectionParameter $parameter, $parameters, $originalParameters)
+    protected function transformDependency(ReflectionParameter $parameter, $parameters)
     {
-        $class = $parameter->getClass();
+        // If the current parameter shouldn't be a class this method will return null
+        if ($class = $parameter->getClass()) {
+
+            // If the parameter has a type-hinted class, we will check to see if it is already in
+            // the list of parameters. If it is we return that class; otherwise, we resolve it here.
+            if ($this->alreadyInParameters($class->name, $parameters)) {
+                return $this->fetchParameterByClassName($class->name, $parameters);
+            }
 
-        // If the parameter has a type-hinted class, we will check to see if it is already in
-        // the list of parameters. If it is we will just skip it as it is probably a model
-        // binding and we do not want to mess with those; otherwise, we resolve it here.
-        if ($class && ! $this->alreadyInParameters($class->name, $parameters)) {
             return $this->container->make($class->name);
         }
     }
 
     /**
+     * Remove known classes from an array of parameters.
+     *
+     * @param  array $originalParameters
+     * @param  array $classNames
+     * @return array
+     */
+    protected function cleanDependenciesFromParameters($originalParameters, $classNames)
```

Comment:
Also here you can add type hinting: `protected function cleanDependenciesFromParameters(array $originalParameters, array $classNames)`